src/Cli/Microsoft.DotNet.Cli.Utils/TypoCorrection.cs

```diff
@@ -110,7 +168,25 @@ private static int GetDistance(string first, string second)
 
             // Return the computed edit distance
             return rows[curRow][m];
+        }
+
+        internal class Suggestion
+        {
+            public Suggestion(string possibleMatch, int distance)
+            {
+                PossibleMatch = possibleMatch;
+                Distance = distance;
+            }
+
+            public string PossibleMatch { get; }
+
+            public int Distance { get; }
+
+            public override bool Equals(object? obj) => obj is Suggestion suggestion && Equals(suggestion);
+
+            public bool Equals(Suggestion other) => string.Equals(PossibleMatch, other.PossibleMatch, StringComparison.OrdinalIgnoreCase);
```

Comment:
nit: This is the only case-insensitive code in this PR, all other comparisons are case-sensitive. Is it needed?